examples/demo-simple-stf/src/lib.rs

```diff
@@ -49,7 +49,7 @@ impl<VM: Zkvm> StateTransitionFunction<VM> for CheckHashPreimageStf {
     // The core logic of our rollup.
     fn apply_blob(
         &mut self,
-        blob: &mut impl BlobTransactionTrait,
+        blob: &mut B,
         _misbehavior_hint: Option<Self::MisbehaviorProof>,
     ) -> BatchReceipt<Self::BatchReceiptContents, Self::TxReceiptContents> {
```

Comment:
I am trying to understand why we need it. The `SequencerOutcome` keeps `sequencer_da_address: Vec<u8>`   Are we going to pass it an `Address` at some point?

Reply:
+Have you thought about using `Box<dyn Address>`? This way, we could avoid passing another generic (not saying it is a better option, just something to consider).

Reply:
> I am trying to understand why we need it. The SequencerOutcome (Self::BatchReceiptContents) keeps sequencer_da_address: Vec<u8> Are we going to pass it an Address at some point?

Yes, idea is to take address from BlobTransactionTrait and keep type instead of converting to bytes via `as_ref()` 

```
-  let sequencer_da_address = blob.sender().as_ref().to_vec();
+  let sequencer_da_address = blob.sender().clone();
```

> Have you thought about using Box<dyn Address> ? This way, we could avoid passing another generic (not saying it is a better option, just something to consider).

It does not work:

```   
error[E0038]: the trait `AddressTrait` cannot be made into an object
   --> module-system/sov-modules-stf-template/src/app_template.rs:31:35
    |
31  |         sequencer_da_address: Box<dyn AddressTrait<Error = anyhow::Error>>,
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `AddressTrait` cannot be made into an object
    |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
   --> /Users/ngolub/workspace/work/sovereign/sovereign/rollup-interface/src/state_machine/mod.rs:17:5
    |
17  |     PartialEq
    |     ^^^^^^^^^ the trait cannot be made into an object because it uses `Self` as a type parameter
    |
   ::: /Users/ngolub/.rustup/toolchains/nightly-2023-02-25-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/cmp.rs:282:15
    |
282 | pub trait Eq: PartialEq<Self> {
    |               ^^^^^^^^^^^^^^^ the trait cannot be made into an object because it uses `Self` as a type parameter
```    


and this approach follows hiding **already added generic**.